testing/unit_tests/t_utils.py

```diff
@@ -38,3 +49,19 @@ def create_tmp_dir(copy_data_testing_dir=True):
     if Path(path_data_testing_source).exists() and copy_data_testing_dir:
         shutil.copytree(path_data_testing_source,
                         path_data_testing_tmp)
+
+
+def create_tmp_dir_multi_session():
+    """Create a temporary directory for data related to multi-session unit tests and copy test data files.
+    1. Remove the ``tmp`` directory if it exists.
```

Comment:
Just to be sure, this "tmp" folder is the same as the one for `create_tmp_dir`?
If that is the case, I assume the tests are run sequentially and the 2 "tmp" folders are not in competition.

Reply:
That is actually a very good point. 

So the current way is like how we are handling functions data testing vs unit data testing: 
![image](https://user-images.githubusercontent.com/6720068/155933856-b8bc1519-83ad-4057-a816-86d861cdc856.png)

A new sub folder is created under `tmp` (for any of the functional or unit) tests. They are all under the same "tmp" folder but in different sub folders. However, if we start parallelerize the tests, they WILL fail because one test finish, nuke the tmp folder, will cause the others to fail. 

This may not be a desired behaviour and we may need to address it in the future if we plan to parallelize the tests